internal/generate/scaffold.go

```diff
@@ -0,0 +1,95 @@
+package generate
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+//go:embed scaffold
+var scaffoldFS embed.FS
+
+type ScaffoldData struct {
+	AppName        string
+	DeploymentName string
+	Description    string
+	Environment    string
+	Namespace      string
+}
+
+func AppScaffold(appName string, environments []string, outputPath string) error {
+	for _, env := range environments {
+		files, err := generateAppScaffoldEnvironmentFiles(appName, env)
+		if err != nil {
+			return err
+		}
+
+		if err := files.Save(filepath.Join(outputPath, env)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateAppScaffoldEnvironmentFiles(appName, environment string) (*Files, error) {
```

Comment:
Would you mind writing a unit test for this? 

It seems there are a few things about this function below that might panic, like accessing boundaries without checks (`s[1:]`), stopping on a directory and more.

Reply:
I've added a golden files test. Originally, I'd considered the `panic` chances low because it comes from an embedded file system, but you're probably right.

This also checks the file content is generated correctly too.